src/makeCalculator.js

```diff
@@ -4,7 +4,23 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  return {
+    result: 0,
+    operate: function (sv, x) {
+      this.result = sv(this.result, x);
+
+      return this;
+    },
+    reset: function () {
+      this.result = 0;
+
+      return this;
+    },
+    multiply: (result, x) => result * x,
+    divide: (result, x) => (x === 0 ? result : result / x),
+    add: (result, x) => result + x,
```

Comment:
Change the `add` method to a regular function to ensure it has access to the calculator's `this` context.